containerd-shim/process.go

```diff
@@ -186,23 +210,46 @@ func (p *process) create() error {
 		}
 		return err
 	}
-	p.stdio.stdout.Close()
-	p.stdio.stderr.Close()
+	if runtime.GOOS != "solaris" { // Temporary hack due to issues with pipes
+		p.stdio.stdout.Close()
+		p.stdio.stderr.Close()
+	}
 	if err := cmd.Wait(); err != nil {
 		if _, ok := err.(*exec.ExitError); ok {
 			return errRuntime
 		}
 		return err
 	}
-	data, err := ioutil.ReadFile("pid")
-	if err != nil {
-		return err
-	}
-	pid, err := strconv.Atoi(string(data))
-	if err != nil {
-		return err
+	if runtime.GOOS == "solaris" {
+		args := append([]string{
+			"--log", logPath,
+			"--log-format", "json",
+		}, p.state.RuntimeArgs...)
+		args = append(args, "start", p.id)
```

Comment:
Solaris does support the split create/start behavior. On Solaris the lifecycke of a container is maintained by a daemon which is orthogonal to the application process of the conrtainer that runs /bin/bash for example. It is only on start that we fork and exec the application process and a pid gets assigned. 

So, yes the call to runz start here is because the logic in the code expects a pid at the end of p.create which on Solaris will be possible only after start.